src/Framework/Constraint/JsonMatches.php

```diff
@@ -45,12 +46,12 @@ public function __construct($value)
      */
     protected function matches($other)
     {
-        list($error, $recodedOther) = $this->canonicalizeJson($other);
+        list($error, $recodedOther) = Json::canonicalize($other);
```

Comment:
One question @evgpisarchik: wouldn't it be better to use non-static methods? I don't know if some kind of injection is possible in this context..

Reply:
According to the project's architecture [Utils](https://github.com/sebastianbergmann/phpunit/tree/master/src/Util) rely on invokation of static methods.

I made my solution consistent with current code base

Reply:
but if you have a proposal on how to refactor current implementation of all Utils let me know please

Reply:
It was just a question, because static access is known to hide the existing dependencies between classes/objects and coupling them in an undesirable way.

Anyway, it looks fine to me if it follows the existent Utils architecture and I don't know the code enough to have a refactor suggestion 😝